### PR TITLE
[em/physics] Add Bethe-Heitler process

### PR DESCRIPTION
**Pull request by vpascuzz**
This PR adds the Bethe-Heitler process for gamma -> e+e-, largely based on Sec. 6.5 of the Geant 4 Physics Reference v10.6.

The following new files have been added:
  - src/physics/em/BetheHeitlerInteractor.hh
  - src/physics/em/BetheHeitlerInteractor.i.hh
  - src/physics/em/BetheHeitlerInteractorPointers.hh
  - src/physics/em/Material.mock.hh
  - test/physics/em/BetheHeitlerInteractor.test.cc
    
and updates test/CMakeLists.txt to include the test (in progress).
    
Note that Material.mock.hh is an over simplification of a material type, and is used here simply for testing purposes.

### CODE REVIEW

File: src/physics/em/BetheHeitlerInteractor.i.hh
Comment (sethrj) [inline src/physics/em/BetheHeitlerInteractor.i.hh:156]: The `allocate.capacity()` here is the number of *all* secondaries that can possibly be stored by the container. Here you should definitely be using `2`. Also, instead of sampling and storing the direction (lines 142–145), just calculate it as a temporary variable in this loop: I think it'll make the code clearer.

File: src/physics/em/BetheHeitlerInteractor.i.hh
Comment (sethrj) [inline src/physics/em/BetheHeitlerInteractor.i.hh:174]: Just as a note for later when we replace the mock material with the material track view and element view: there will be a `element.cbrt_z()` so you can can preferentially use that. There are a couple of other places that will need similar updates.

File: src/physics/em/BetheHeitlerInteractor.i.hh
Comment (sethrj) [inline src/physics/em/BetheHeitlerInteractor.i.hh:232]: `3 / 2 * x` is `1 * x` is `x` -- write as `1.5` or declare a local `constexpr real_type three_halves = 3.0 / 2;`.

File: src/physics/em/BetheHeitlerInteractor.i.hh
Comment (sethrj) [inline src/physics/em/BetheHeitlerInteractor.i.hh:203]: `8 / 3 == 2`.

**Review by sethrj — CHANGES_REQUESTED** [review]
Looking really good, Vince: I spotted a couple of math type errors and made a few other comments about converting formulas into code. I think the element view interface is on the `hackathon` branch, so if you can I'd switch your `mock` stuff to that.

**Comment by vpascuzz** [comment]
Closed with reason?

**Comment by vpascuzz** [comment]
Apologies if I missed something!

**Comment by sethrj** [comment]
Ugh sorry @vpascuzz , this was an unintended side effect from deleting the upstream `physics-hackathon` branch, which I realize was the target branch for this MR. Should I restore that branch?  Or would it make more sense to start breaking up our "physics hackathon" branches away from the original skeleton commit I made? I am inclined toward the latter, but let's wait until the material branch #59 gets merged.

**Pull request closed without merging**
